aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -574,15 +986,74 @@ module aptos_framework::delegation_pool {
     public entry fun set_delegated_voter(
         owner: &signer,
         new_voter: address
-    ) acquires DelegationPoolOwnership, DelegationPool {
+    ) acquires DelegationPoolOwnership, DelegationPool, GovernanceRecords {
+        // No one can change delegated_voter once the partial governance voting feature is enabled.
+        assert!(!features::delegation_pool_partial_governance_voting_enabled(), error::invalid_argument(EDEPRECATED_FUNCTION));
         let pool_address = get_owned_pool_address(signer::address_of(owner));
         // synchronize delegation and stake pools before any user operation
         synchronize_delegation_pool(pool_address);
         stake::set_delegated_voter(&retrieve_stake_pool_owner(borrow_global<DelegationPool>(pool_address)), new_voter);
     }
 
+    /// Allows a delegator to delegate its voting power to a voter. If this delegator already has a delegated voter,
+    /// this change won't take effects until the next lockup period.
+    public entry fun delegate_voting_power(
+        delegator: &signer,
+        pool_address: address,
+        new_voter: address
+    ) acquires DelegationPool, GovernanceRecords {
+        assert_partial_governance_voting_enabled(pool_address);
+
+        // synchronize delegation and stake pools before any user operation
+        synchronize_delegation_pool(pool_address);
+
+        let delegator_address = signer::address_of(delegator);
+        let (active, inactive, pending_inactive) = get_stake(pool_address, delegator_address);
+        assert!(active + inactive + pending_inactive > 0, error::invalid_argument(EDELEGATOR_NOT_EXISTS));
```

Comment:
is this check required? a delegator that never participated in the delegation pool might be a dApp calling this at its initialization for example

Reply:
My concern was that people might create tons of addresses and call this function. But maybe not a vulnerability.